features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -813,7 +804,10 @@ void ATHandler::resp_start(const char *prefix, bool stop)
 
     // Try get as much data as possible
     rewind_buffer();
-    fill_buffer();
+    if (!fill_buffer()) {
+        tr_error("fill failed %s", prefix);
+        return;
```

Comment:
What if we have already needed data in buffer, fetched during process_oob() -> match_urc() -> set_scope(InfoType) -> consume_char(' ') -> get_char() -> fill_buffer()? Should return only if buffer is also empty?

Reply:
`fill_buffer` is currently waiting for data. Which it should only be doing if it thinks it hasn't got enough. So presumably it shouldn't be being called at all here. Or it should be being called with a zero timeout, at least, if you're trying to get stuff out of the serial port.

`resp` a bit later does a `fill_buffer` when it's out of data - that's the one that should have a timeout for the AT response time.